ios/CoinbaseWalletSDK/Provider/MobileWalletProvider.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  WalletProvider.swift
+//  CoinbaseWalletSDK
+//
+//  Created by Amit Goel on 11/4/22.
+//
+
+public protocol MobileWalletProviderProtocol {
+    func getWallets() -> [Wallet]
+}
+
+public class MobileWalletProvider : MobileWalletProviderProtocol {
+    
+    public init() {}
+    
+    public func getWallets() -> [Wallet] {
+        return [.coinbaseWallet, .coinbaseRetail]
+    }
+}
+
+extension Wallet {
+    public static let coinbaseWallet = Wallet(name: "Coinbase Wallet",
+                                              iconUrl: "https://is1-ssl.mzstatic.com/image/thumb/Purple122/v4/bd/2b/dc/bd2bdcac-44ff-0707-3ec1-fde5014a91a1/AppIcon-0-0-1x_U007emarketing-0-0-0-10-0-0-sRGB-0-0-0-GLES2_U002c0-512MB-85-220-0-0.png/100x100bb.jpg",
+                                              url: "cbwallet://wsegue",
+                                              appStoreUrl: "https://apps.apple.com/us/app/coinbase-wallet-nfts-crypto/id1278383455")
```

Comment:
this should use universal link
+ nit: swift convention

```suggestion
    public static let coinbaseWallet = Wallet(
        name: "Coinbase Wallet",
        iconUrl: "https://is1-ssl.mzstatic.com/image/thumb/Purple122/v4/bd/2b/dc/bd2bdcac-44ff-0707-3ec1-fde5014a91a1/AppIcon-0-0-1x_U007emarketing-0-0-0-10-0-0-sRGB-0-0-0-GLES2_U002c0-512MB-85-220-0-0.png/100x100bb.jpg",
        url: "https://........",
        appStoreUrl: "https://apps.apple.com/us/app/coinbase-wallet-nfts-crypto/id1278383455"
)
```